### PR TITLE
#3 update pom.xml

### PR DESCRIPTION
**Pull request by handk85**
This commit updates `pom.xml` in line with COMP0010 coding standards.

Also, it removes several Checkstyle issues.

### CODE REVIEW

File: LectureDemo/pom.xml [hunk 1]
```diff
@@ -43,48 +44,122 @@
...
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-javadoc-plugin</artifactId>
+        <version>3.10.0</version>
+        <configuration>
+          <doclint>all,-missing</doclint>
+        </configuration>
+      </plugin>
+    </plugins>
+  </reporting>
```
Comment (handk85) [inline LectureDemo/pom.xml:164]: This part was copied and paste from the document.

File: LectureDemo/src/main/java/uk/ac/ucl/comp0010/model/EmptyRateException.java [hunk 2]
```diff
@@ -1,5 +1,8 @@
 package uk.ac.ucl.comp0010.model;
 
+/**
+ * An exception when there is no rate.
+ */
```
Comment (handk85) [inline LectureDemo/src/main/java/uk/ac/ucl/comp0010/model/EmptyRateException.java:5]: This Javadoc was added to resolve Chekcstyle Issues.

**Review by handk85 — COMMENTED** [review]
It looks good to me.

**Review by levon-d — APPROVED** [review]
lgtm!

**Comment by Declan-Loo** [comment]
LGTM!!!!

**Pull request merged**